File: 문제풀이/프로그래머스/코딩 기초 트레이닝/Level 0. 홀짝에 따라 다른 값 반환하기.js
function solution(n) {
  return Array.from({ length: n }, (_, v) => v + 1)
    .filter((v) => (n % 2 ? v % 2 : !(v % 2)))
    .map((v) => (n % 2 ? v : v ** 2))
    .reduce((a, b) => a + b);
}

File: 문제풀이/프로그래머스/코딩 기초 트레이닝/Level 0. 홀짝에 따라 다른 값 반환하기.py
def solution(n: int) -> int:
    if n % 2:
        return sum(i for i in range(1, n + 1, 2))
    else:
        return sum(i * i for i in range(2, n + 1, 2))


print(solution(7))
print(solution(10))
